dags/multipod/configs/legacy_unit_test.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to run legacy tests from the old infrastructure."""
+
+import datetime
+import os
+from xlml.apis import gcp_config, metric_config, task, test_config
+from collections.abc import Iterable
+from dags import test_owner
+from dags.multipod.configs import common
+from dags.vm_resource import TpuVersion, Project, RuntimeVersion, ClusterName
+
+
+def get_legacy_unit_test_config(
+    test_cmd: Iterable[str],
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    test_mode: common.SetupMode,
+    test_owner: str,
+    docker_image: str,
+    num_slices: int = 1,
+    cluster_name: str = ClusterName.V4_8_MULTISLICE_CLUSTER.value,
+    project_name: str = Project.TPU_PROD_ENV_MULTIPOD.value,
+) -> task.TpuXpkTask:
+  """
+  Run a legacy unit test script.
+  test_cmd will be executed within the `dags/multipod/legacy_tests` directory.
+  """
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.XLML_DATASET,
+  )
+
+  run_model_cmds = (
+      f'git clone https://github.com/GoogleCloudPlatform/ml-auto-solutions',
```

Comment:
I'm a little concerned this method makes this type of test hard to test. Any changes have to get pushed to the repository before you can do a test run, so it would really be better if the test case file originated from the `dags` folder at runtime.

Could you instead embed the file in the setup commands?

Reply:
Sadly it would be messy... I've thought about this some already:

-  `xpk` tests run in an isolated environment from the airflow worker, so there's no way to have direct access to `dags` or SCP it over to the workload.
- There can't be any newlines in the workload command, so we'd need to encode the test script to embed it in the workload command (messy). 
- `xpk` also doesn't support configmaps to communicate the script across
- We can't use the `base_docker_image` feature to bake it into the image, since docker build would fail within the composer env.

It's unfortunate, but this is the existing pattern in the internal repo as well. cc @Obliviour in case there's an XPK feature I'm overlooking.

Reply:
> There can't be any newlines in the workload command, so we'd need to encode the test script to embed it in the workload command (messy).

Python has a built-in function to escape strings including quotes, newlines, etc: https://docs.python.org/3/library/shlex.html#shlex.quote

You should be able to use that unless the XPK script is doing something funky with the input.

Reply:
It looks like `shlex.quote` doesn't replace newlines (unless I'm missing something), so we'd have to manage that separately. It does easily work with base64 encoding, but it's pretty gross... https://paste.googleplex.com/6607797074001920